lib/std/os/linux/test.zig

```diff
@@ -75,3 +75,18 @@ test "statx" {
     expect(@bitCast(u64, @as(i64, stat_buf.blksize)) == statx_buf.blksize);
     expect(@bitCast(u64, @as(i64, stat_buf.blocks)) == statx_buf.blocks);
 }
+
+test "getrusage" {
```

Comment:
Could you move this test to `os/test.zig` and go through the new `std.os` API that you just introduced? Our test suite tests versions of std lib with no libc, musl libc, glibc, macos libc, and freebsd libc, and so this will provide significantly more coverage.

This will require adding support for other operating systems, such as FreeBSD and macOS, which I do consider to be part of this PR, because it will inform whether the API can possibly have an error or not.

If you grow tired of working on this, let me know, I will be happy to complete the branch and merge it.

Reply:
> If you grow tired of working on this

No! But I have to go to bed. :)

Can you write test, please?

Reply:
Sure. Let me know when you are ready to hand off the PR.

Reply:
I hope PR is ready.
Just an idea: convert ```RUSAGE_*``` consts to ```RUsageWho``` enum.
My next PR depends on this, so I'll wait... :)